src/joystick/hidapi/SDL_hidapi_flydigi.c

```diff
@@ -0,0 +1,408 @@
+/*
+  Simple DirectMedia Layer
+  Copyright (C) 1997-2024 Sam Lantinga <slouken@libsdl.org>
+
+  This software is provided 'as-is', without any express or implied
+  warranty.  In no event will the authors be held liable for any damages
+  arising from the use of this software.
+
+  Permission is granted to anyone to use this software for any purpose,
+  including commercial applications, and to alter it and redistribute it
+  freely, subject to the following restrictions:
+
+  1. The origin of this software must not be misrepresented; you must not
+     claim that you wrote the original software. If you use this software
+     in a product, an acknowledgment in the product documentation would be
+     appreciated but is not required.
+  2. Altered source versions must be plainly marked as such, and must not be
+     misrepresented as being the original software.
+  3. This notice may not be removed or altered from any source distribution.
+*/
+#include "SDL_internal.h"
+
+#ifdef SDL_JOYSTICK_HIDAPI
+
+#include "../SDL_sysjoystick.h"
+#include "SDL_hidapijoystick_c.h"
+#include "SDL_hidapi_rumble.h"
+
+#ifdef SDL_JOYSTICK_HIDAPI_FLYDIGI
+
+// Define this if you want to log all packets from the controller
+#if 0
+#define DEBUG_FLYDIGI_PROTOCOL
+#endif
+
+enum
+{
+    SDL_GAMEPAD_BUTTON_FLYDIGI_M1 = 11,
+    SDL_GAMEPAD_BUTTON_FLYDIGI_M2,
+    SDL_GAMEPAD_BUTTON_FLYDIGI_M3,
+    SDL_GAMEPAD_BUTTON_FLYDIGI_M4,
+    SDL_GAMEPAD_BUTTON_FLYDIGI_C,
+    SDL_GAMEPAD_BUTTON_FLYDIGI_Z,
+    SDL_GAMEPAD_NUM_FLYDIGI_BUTTONS,
+};
+
+#define FLYDIGI_ACCEL_SCALE 256.f
```

Comment:
```suggestion
#define FLYDIGI_ACCEL_SCALE 5120.f
```
This seems like the correct scale for the Apex 3 controller. The accelerometer axes seem correct. I can double check the gyro once this is merged.

Are the internal sensors updated at 125 Hz?

Reply:
No idea, I just tuned the value to get 9.8 while my vader4 is laying flat

Reply:
Tried to do the same with gyro comparing with the values from dualsense, but confidence is much lower there.

Reply:
Device name should be enough
Though looks like what is different are buttons bitmasks

Reply:
Yes, buttons are working correctly, though there's no C/Z button. We probably should dynamically set the number of available buttons based on the button bitmask.